File: node-aws-whoami-v3/whoami.mjs
import { fromNodeProviderChain } from "@aws-sdk/credential-providers";
import { STSClient, GetCallerIdentityCommand } from "@aws-sdk/client-sts";

function providerLogger() {
    // remember last logging module name
    let last = "";
    return {
        debug: (str) => last = str,
        getProvider: () => last,
    };
}

async function whoami() {
    const logger = providerLogger();
    const client = new STSClient({
        credentials: fromNodeProviderChain({ logger }),
    });
    const command = new GetCallerIdentityCommand({});
    const response = await client.send(command);
    console.log("+++ AWS whoami +++");
    console.log(`Credentials type: ${logger.getProvider()}`);
    console.log(`User ID: ${response.UserId}`);
    console.log(`Account: ${response.Account}`);
    console.log(`ARN: ${response.Arn}`);
};

await whoami();

File: py3-aws-whoami/whoami.py
#!/usr/bin/env python3

import boto3

client = boto3.client('sts')

def whoami():
    response = client.get_caller_identity()
    print('+++ AWS whoami +++')
    print('User ID: %s' % response['UserId'])
    print('Account: %s' % response['Account'])
    print('ARN: %s' % response['Arn'])

if __name__ == '__main__':
    whoami()
